scripts/build_shapes.py

```diff
@@ -248,26 +318,21 @@ def eez(countries, geo_crs, country_shapes, EEZ_gpkg, out_logging=False, distanc
     # load data
     df_eez = load_EEZ(countries, geo_crs, EEZ_gpkg)
 
-    ret_df = df_eez[["name", "geometry"]]
-    # create unique shape if country is described by multiple shapes
-    for c_code in countries:
-        selection = ret_df.name == c_code
-        n_offshore_shapes = selection.sum()
-
-        if n_offshore_shapes > 1:
-            # when multiple shapes per country, then merge polygons
-            geom = ret_df[selection].geometry.unary_union
-            ret_df.drop(ret_df[selection].index, inplace=True)
-            ret_df = ret_df.append(
-                {"name": c_code, "geometry": geom}, ignore_index=True
-            )
+    ret_df = gpd.GeoDataFrame(
+        {
+            "name": df_eez.name.copy(),
+            "geometry": [
+                df_eez.geometry.loc[df_eez.name == cc].geometry.unary_union
+                for cc in df_eez.name
```

Comment:
An elegant replacement. But wouldn't it lead to a mismatch when there are multiple rows for a single country?

Reply:
Nice spotting! Revised